lib/buffer.js

```diff
@@ -151,30 +150,20 @@ function fromObject(obj) {
   }
 
   if (obj.buffer instanceof ArrayBuffer || obj.length) {
```

Comment:
Not very familiar with node/js internal, but I think the in operator should not be very slow compare to Array.isArray?

PR updated according to your comments